### PR TITLE
cmake: support sanitizers natively

### PR DESCRIPTION
**Pull request by lzaoral**

### CODE REVIEW

File: tests/0010-csexec-readlink/runtest.sh [hunk 1]
```diff
@@ -11,6 +11,7 @@ gcc -Wl,--dynamic-linker="${PATH_TO_CSEXEC_LOADER}" -o out \
 
 # run
 export LD_LIBRARY_PATH="${PATH_TO_CSEXEC_LIBS}"
-./out 2> stderr.txt | grep -vsE '\/ld[^\/]*\.so\.[[:digit:]]$' || exit 1
+./out > stdout.txt 2> stderr.txt
 
+grep -vsE '\/ld[^\/]*\.so\.[[:digit:]]$' stdout.txt || { cat stdout.txt; exit 1; }
```
Comment (kdudka) [inline tests/0010-csexec-readlink/runtest.sh:18]: Should `stderr` be compared first as the commit message says?

**Review by kdudka — APPROVED** [review]
Looks good.  Thanks!  Some minor comments inline...

File: tests/0008-csexec-basic/runtest.sh [hunk 2]
```diff
@@ -6,8 +6,11 @@ set -x
 gcc -Wl,--dynamic-linker="${PATH_TO_CSEXEC_LOADER}" -o out \
     "${TEST_SRC_DIR}/test.c"
 
-export LD_LIBRARY_PATH="${PATH_TO_CSEXEC_LIBS}"
-CSEXEC_WRAP_CMD="${TEST_SRC_DIR}/wrap.sh" ./out > stdout.txt 2> stderr.txt
+# run
+CSEXEC_WRAP_CMD="${TEST_SRC_DIR}/wrap.sh" \
+LD_LIBRARY_PATH="${PATH_TO_CSEXEC_LIBS}"  \
+./out > stdout.txt 2> stderr.txt
```
[the remark this answers is not in the document]
Comment (lzaoral) [reply tests/0008-csexec-basic/runtest.sh]: Addressed in 2f94d6d and 17195b3.

File: tests/0010-csexec-readlink/runtest.sh [hunk 3]
```diff
@@ -10,8 +10,9 @@ gcc -Wl,--dynamic-linker="${PATH_TO_CSEXEC_LOADER}" -o out \
     "${TEST_SRC_DIR}/test.c"
 
 # run
-export LD_LIBRARY_PATH="${PATH_TO_CSEXEC_LIBS}"
+LD_LIBRARY_PATH="${PATH_TO_CSEXEC_LIBS}" \
 ./out > stdout.txt 2> stderr.txt
```
[the remark this answers is not in the document]
Comment (lzaoral) [reply tests/0010-csexec-readlink/runtest.sh]: Addressed in 2f94d6d and 17195b3.

File: tests/testlib.sh [hunk 4]
```diff
@@ -1,3 +1,5 @@
+# shellcheck shell=bash
```
[the remark this answers is not in the document]
Comment (lzaoral) [reply tests/testlib.sh:1]: Thanks, that was a typo. Fixed in 4fc38d7.

File: tests/testlib.sh [hunk 5]
```diff
@@ -20,12 +20,55 @@ PATH_TO_CSEXEC_LIBS="$(dirname "$3")"
...
+
+    export LIBASAN_PATH
+fi
+
+check_ld_so_flag() {
+    (
+        LD_PRELOAD="$LIBASAN_PATH" \
+        "$PATH_TO_CSEXEC_LIBS/csexec" --print-ld-exec-cmd argv0 | grep -qv -- "$1"
+    )
+}
```
[the remark this answers is not in the document]
Comment (lzaoral) [reply tests/testlib.sh:62]: Addressed in 17195b3.

**Pull request merged**